server/buildbuddy_server/buildbuddy_server.go

```diff
@@ -886,6 +888,48 @@ func (s *BuildBuddyServer) GetUsage(ctx context.Context, req *usagepb.GetUsageRe
 	return nil, status.UnimplementedError("Not implemented")
 }
 
+func (s *BuildBuddyServer) GetCacheMetadata(ctx context.Context, req *capb.GetCacheMetadataRequest) (*capb.GetCacheMetadataResponse, error) {
+	ctx, err := prefix.AttachUserPrefixToContext(ctx, s.env)
+	if err != nil {
+		return nil, err
+	}
+
+	resourceName := req.GetResourceName()
+	cacheType, err := ProtoCacheTypeToCacheType(resourceName.GetCacheType())
+	if err != nil {
+		return nil, err
+	}
+	cache, err := s.env.GetCache().WithIsolation(ctx, cacheType, resourceName.GetInstanceName())
+	if err != nil {
+		return nil, err
+	}
+
+	metadata, err := cache.Metadata(ctx, resourceName.GetDigest())
```

Comment:
does calling Metadata() update the last access time? (if so, do we need a flag in the request that tells it not to update?)

Reply:
No it doesn't